evm_loader/ERC20/test/test.py

```diff
@@ -138,383 +44,418 @@ def deployERC20(self, location_hex, location_bin,  mintId, balance_erc20):
                                        AccountMeta(pubkey=self.caller, is_signer=False, is_writable=True),
                                        AccountMeta(pubkey=PublicKey(sysinstruct), is_signer=False, is_writable=False),
                                        AccountMeta(pubkey=self.loader.loader_id, is_signer=False, is_writable=False),
                                        AccountMeta(pubkey=PublicKey(sysvarclock), is_signer=False, is_writable=False),
                                    ]))
-        result = http_client.send_transaction(trx, self.acc)
-        result = confirm_transaction(http_client, result["result"])
+        result = send_transaction(client, trx, self.acc)
         messages = result["result"]["meta"]["logMessages"]
-        print("erc20 transfer signature: {}".format(result["result"]["transaction"]["signatures"][0]))
-        res = messages[messages.index("Program log: succeed") + 1]
-        if not res.startswith("Program log: "):
-            raise Exception("Invalid program logs: no result")
+        res = messages[-1]
+        print("res:", res)
+        if any(search("Program %s failed" % evm_loader_id, m) for m in messages):
+            raise Exception("Invalid program logs: Program %s failed" % evm_loader_id)
         else:
-            if int(res[13:], 16) == 1:
-                print("transfer OK")
-            else:
-                print("transfer Fail")
+            return res
 
-    def erc20_balance_ext(self, erc20):
+    def erc20_balance_ext(self, erc20, erc20_code):
         input = bytearray.fromhex("0340b6674d")
         trx = Transaction().add(
             TransactionInstruction(program_id=self.loader.loader_id, data=input, keys=
             [
                 AccountMeta(pubkey=erc20, is_signer=False, is_writable=True),
+                AccountMeta(pubkey=erc20_code, is_signer=False, is_writable=True),
                 AccountMeta(pubkey=self.caller, is_signer=False, is_writable=True),
                 AccountMeta(pubkey=self.acc.public_key(), is_signer=True, is_writable=False),
                 AccountMeta(pubkey=self.loader.loader_id, is_signer=False, is_writable=False),
                 AccountMeta(pubkey=PublicKey(sysvarclock), is_signer=False, is_writable=False),
             ]))
 
-        result = http_client.send_transaction(trx, self.acc)
-        result = confirm_transaction(http_client, result["result"])
+        result = send_transaction(client, trx, self.acc)
         messages = result["result"]["meta"]["logMessages"]
-        res = messages[messages.index("Program log: succeed") + 1]
-        if not res.startswith("Program log: "):
-            raise Exception("Invalid program logs: no result")
+        res = messages[-1]
+        print("balance_ext", res)
+        if any(search("Program %s failed" % evm_loader_id, m) for m in messages):
+            raise Exception("Invalid program logs: Program %s failed" % evm_loader_id)
         else:
```

Comment:
this is unnecessary.
please , remove it

Reply:
Fixed